src/meta/meta_bulk_load_ingestion_context.cpp

```diff
@@ -0,0 +1,204 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+#include "meta_bulk_load_ingestion_context.h"
+
+#include <dsn/dist/fmt_logging.h>
+#include <dsn/utility/fail_point.h>
+
+namespace dsn {
+namespace replication {
+
+DSN_DEFINE_uint32("meta_server",
+                  bulk_load_node_max_ingesting_count,
+                  4,
+                  "max partition_count executing ingestion for one node at the same time");
+DSN_TAG_VARIABLE(bulk_load_node_max_ingesting_count, FT_MUTABLE);
+
+DSN_DEFINE_uint32("meta_server", bulk_load_node_min_disk_count, 1, "min disk count of one node");
+DSN_TAG_VARIABLE(bulk_load_node_min_disk_count, FT_MUTABLE);
+
+ingestion_context::ingestion_context() { reset_all(); }
+
+ingestion_context::~ingestion_context() { reset_all(); }
+
+// ingestion_context::partition_node_info
+void ingestion_context::partition_node_info::create(const partition_configuration &config,
+                                                    const config_context &cc)
+{
+    pid = config.pid;
```

Comment:
I found that pid is useless

Reply:
It just like a name of `partition_node_info` to show which partition it is.

Reply:
What I mean is in this function, the variable named pid is useless. I think you got a mistake

Reply:
Ohh, this function name is `create`, just used to initialize structure `partition_node_info`, so it will init all members of it.